### ISSUE TITLE
Not auto skipping in Safari

### ISSUE THREAD

**Issue by planet0** [ajayyy/SponsorBlock#943]
Hello, I've recently moved from Google Chrome on Windows to Safari on macOS and noticed that video segments are not auto skipping at random times, but on Google Chrome it would auto skip perfectly.

It seems to mostly occur when there's a segment to skip right at the start of the video.

<details><summary>Debug Info</summary>

{
    "debug": {
        "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.2 Safari/605.1.15",
        "platform": "MacIntel",
        "language": "en-au",
        "extensionVersion": "3.1.1"
    },
    "config": {
        "showTimeWithSkips": false,
        "defaultCategory": "chooseACategory",
        "hideDeleteButtonPlayerControls": false,
        "checkForUnlistedVideos": false,
        "trackViewCount": true,
        "showDonationLink": false,
        "supportInvidious": false,
        "noticeVisibilityMode": 3,
        "trackViewCountInPrivate": true,
        "serverAddress": "Default server address",
        "whitelistedChannels": 0,
        "ytInfoPermissionGranted": false,
        "allowExpirements": true,
        "disableSkipping": false,
        "skipNoticeDuration": "5",
        "categorySelections": [
            {
                "name": "sponsor",
                "option": 2
            },
            {
                "name": "selfpromo",
                "option": 2
            },
            {
                "name": "interaction",
                "option": 2
            },
            {
                "name": "poi_highlight",
                "option": 0
            },
            {
                "name": "intro",
                "option": 2
            },
            {
                "name": "outro",
                "option": 2
            },
            {
                "name": "preview",
                "option": 2
            },
            {
                "name": "music_offtopic",
                "option": 1
            }
        ],
        "highlightCategoryUpdate": true,
        "hideDiscordLink": false,
        "minDuration": 0,
        "segmentTimes": [
            [
                "x2z-BMvwh7U",
                [
                    {
                        "segment": [
                            10.333814
                        ],
                        "category": "chooseACategory",
                        "UUID": null
                    }
                ]
            ],
            [
                "xMPkgtOxlfo",
                [
                    {
                        "segment": [
                            1628.083106,
                            1644.801
                        ],
                        "actionType": "skip",
                        "UUID": null,
                        "category": "outro",
                        "source": 1
                    }
                ]
            ],
            [
                "U49pq-Ctf_A",
                [
                    {
                        "segment": [
                            652.647121
                        ],
                        "actionType": "skip",
                        "UUID": null,
                        "category": "chooseACategory",
                        "source": 1
                    }
                ]
            ]
        ],
        "refetchWhenNotFound": true,
        "muteSegments": false,
        "skipCount": 1195,
        "forceChannelCheck": true,
        "autoSkipOnMusicVideos": false,
        "previewCategoryUpdate": true,
        "testingServer": false,
        "audioNotificationOnSkip": false,
        "barTypes": {
            "preview-intro": {
                "color": "#008080",
                "opacity": "0.7"
            },
            "preview": {
                "color": "#008fd6",
                "opacity": "0.7"
            },
            "preview-chooseACategory": {
                "color": "#ffffff",
                "opacity": "0.7"
            },
            "intro": {
                "color": "#00ffff",
                "opacity": "0.7"
            },
            "preview-selfpromo": {
                "color": "#bfbf35",
                "opacity": "0.7"
            },
            "poi_highlight": {
                "color": "#ff1684",
                "opacity": "0.7"
            },
            "preview-interaction": {
                "color": "#6c0087",
                "opacity": "0.7"
            },
            "preview-poi_highlight": {
                "color": "#9b044c",
                "opacity": "0.7"
            },
            "music_offtopic": {
                "color": "#ff9900",
                "opacity": "0.7"
            },
            "preview-music_offtopic": {
                "color": "#a6634a",
                "opacity": "0.7"
            },
            "sponsor": {
                "color": "#00d400",
                "opacity": "0.7"
            },
            "preview-outro": {
                "color": "#000070",
                "opacity": "0.7"
            },
            "interaction": {
                "color": "#cc00ff",
                "opacity": "0.7"
            },
            "outro": {
                "color": "#0202ed",
                "opacity": "0.7"
            },
            "preview-preview": {
                "color": "#005799",
                "opacity": "0.7"
            },
            "preview-sponsor": {
                "color": "#007800",
                "opacity": "0.7"
            },
            "selfpromo": {
                "color": "#ffff00",
                "opacity": "0.7"
            }
        },
        "hideInfoButtonPlayerControls": false,
        "highlightCategoryAdded": true,
        "autoHideInfoButton": true,
        "sponsorTimesContributed": 157,
        "submissionCountSinceCategories": 135,
        "submitKeybind": "'",
        "hideVideoPlayerControls": false,
        "invidiousInstances": 1,
        "hideDiscordLaunches": 0,
        "hideUploadButtonPlayerControls": false,
        "autoSkipOnMusicVideosUpdate": true,
        "minutesSaved": 422.50657285000045,
        "dontShowNotice": false,
        "startSponsorKeybind": ";",
        "skipKeybind": "Enter"
    }
}

</details>

**Comment by roman-raizen** [comment]
it's been forever, especially when i updated to Monterey. it dropped from "works sometimes" to "almost never works".

**Comment by ajayyy** [comment]
Double check tgwt safari hasn't removed the YouTube.com permission

**Comment by planet0** [comment]
Permissions still allowed:

<img width="904" alt="Screen Shot 2021-09-22 at 12 04 07 pm" src="https://user-images.githubusercontent.com/65204836/134271984-fbc649c8-ada7-4c68-8ead-bbbac827845e.png">

**Comment by ajayyy** [comment]
Could you check the console for the browser tab next time it happens. I wonder if it is https://stackoverflow.com/questions/53939205/how-to-avoid-extension-context-invalidated-errors-when-messaging-after-an-exte

**Comment by planet0** [comment]
Hi @ajayyy, any update on this? For now I've been quick pausing/resuming with the spacebar to start skipping segments.

**Comment by ajayyy** [comment]
@planet0 DOes quick pausing/resuming immediatly after fix it every time?

**Comment by ajayyy** [comment]
I don't think the incorrect video check can be the cause. I cannot reproduce that when forcing it to fail for the first load.

**Comment by planet0** [comment]
@ajayyy Yes it does fix it every time. Could it be that the video starts playing before SponsorBlock has the chance to even load itself and doesn't realise that the video is already playing? (So pause/unpause makes it realise it)

**Comment by ajayyy** [comment]
Does this occur only on first load, or can it happen to videos played later? (clicking a recommended video doesn't "reload" the page, so SponsorBlock stays loaded)

**Comment by planet0** [comment]
You're right, it's the opposite behaviour - it's only occurring when loading videos via their 'lazy load' and doesn't have a problem on first load.

**Comment by ajayyy** [comment]
4.0 changes how videoID detection works, let me know if this still occurs

**Comment by planet0** [comment]
@ajayyy It looks like 4.0 does detect ID's correctly, I now see two of them each time.
```
[SponsorBlock] The videoID recorded when trying to skip is different than what it should be.
[SponsorBlock] VideoID recorded: bC_W93NwQ5A. Actual VideoID: 0d4q6CTY-0M
```

**Comment by ajayyy** [comment]
Going to close this as it's old
